nomos-core/chain-defs/src/proofs/leader_proof.rs

```diff
@@ -118,8 +118,10 @@ mod test {
     fn note_commitment_leaves(
         note_commitments: &[cl::NoteCommitment],
     ) -> [[u8; 32]; MAX_NOTE_COMMS] {
-        let note_comm_bytes =
-            Vec::from_iter(note_commitments.iter().map(|c| c.as_bytes().to_vec()));
+        let note_comm_bytes = note_commitments
+            .iter()
+            .map(|c| c.as_bytes().to_vec())
+            .collect::<Vec<_>>();
```

Comment:
I would rather use the type in the variable declaration rather than collect though.
`let note_comm_bytes: Vec<_> = ...`
Is this linted as well?

Reply:
Yes, I believe that won't trigger warnings. The key here was to replace `::from_iter` with `collect`. I will change that.

Reply:
Let me mention we have a mixed approach throughout our codebase. There is plenty of places in which we use turbofish syntax, so this would not be the only instance. But I changed the approach here: https://github.com/logos-co/nomos-node/pull/1093/commits/20ebe764a10ae25cf4c5777ac7940e6d0db212b1.